src/click/core.py

```diff
@@ -2346,6 +2354,19 @@ def value_is_missing(self, value: t.Any) -> bool:
         return False
 
     def process_value(self, ctx: Context, value: t.Any) -> t.Any:
+        """Process the value of this parameter:
+
+        1. Type cast the value using :meth:`type_cast_value`.
+        2. Check if the value is missing (see: :meth:`value_is_missing`), and raise
+           :exc:`MissingParameter` if it is required.
+        3. If a :attr:`callback` is set, call it to have the value replaced by the
+           result of the callback. The callback is receiving the value as-is, which let
+           the developer decide how to handle the different cases.
+
+        .. versionchanged:: 8.3
+            The :attr:`callback` gets an internal sentinel if the parameter was not set
```

Comment:
This doesn't seem correct? User code should not be interacting with `UNSET`, and command/param callbacks are user code.